File: rb/features.rb
# KiprioHttpApis SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module KiprioHttpApisFeatures
  def self.make_feature(name)
    case name
    when "base"
      KiprioHttpApisBaseFeature.new
    when "test"
      KiprioHttpApisTestFeature.new
    else
      KiprioHttpApisBaseFeature.new
    end
  end
end

File: py/features.py
# KiprioHttpApis SDK feature factory

from feature.base_feature import KiprioHttpApisBaseFeature
from feature.test_feature import KiprioHttpApisTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KiprioHttpApisBaseFeature(),
        "test": lambda: KiprioHttpApisTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
